bookserver/routers/rslogging.py

```diff
@@ -179,3 +188,145 @@ async def runlog(request: Request, response: Response, data: LogRunIncoming):
 async def same_class(user1: AuthUserValidator, user2: str) -> bool:
     u2 = await fetch_user(user2)
     return user1.course_id == u2.course_id
+
+
+# completion tables
+# =================
+#
+# This section contains implementations of endpoints for tracking progress
+
+
+# updatelastpage
+# --------------
+@router.post("updatelastpage")
+async def updatelastpage(request: Request, request_data: LastPageIncoming):
+    if request_data.last_page_url is None:
+        return  # todo:  log request_data, request.args and request.env.path_info
+
+    lpd = LastPageData(**request_data.dict())
+
+    lpd.last_page_chapter = request_data.last_page_url.split("/")[-2]
```

Comment:
Some comments would be nice: "The URL provided must be ``chapter/subchapter.html``. Therefore, this code..."